### PR TITLE
Lerna-ize financial-templates-lib

### PR DESCRIPTION
**Pull request by mrice32**
This adds lerna support to the financial-templates-lib.

Other parts of the codebase import `financial-templates-lib` by calling `require("@umaprotocol/financial-templates-lib")`.

Once truffle releases [this PR](https://github.com/trufflesuite/truffle/pull/3157), we should be able to begin running tests from these directories because `artifacts.require("@umaprotocol/core/ContractName)` should work natively. We could also choose to just manually import the json and construct the contract abstractions manually in these directories. I'm probably just going to wait on the truffle release, but if someone wants to take it upon themselves to do this refactor, go for it.

Notes:
- I added a `files` specifier to the `core` package to ensure that the `contracts` and `build` directories are included in the package. I think, by default, build would be excluded because it's in our .gitignore.
- In the course of running lerna/yarn commands to add packages, some existing `package.json` dependencies were resorted. That's the reason behind the churn in those files.

### CODE REVIEW

File: financial-templates-lib/price-feed/CreatePriceFeed.js [hunk 1]
```diff
@@ -3,7 +3,7 @@ const { MedianizerPriceFeed } = require("./MedianizerPriceFeed");
 const { CryptoWatchPriceFeed } = require("./CryptoWatchPriceFeed");
 const { UniswapPriceFeed } = require("./UniswapPriceFeed");
 
-const Uniswap = require("../../core/build/contracts/Uniswap.json");
+const Uniswap = require("@umaprotocol/core/build/contracts/Uniswap.json");
```
Comment (nicholaspai) [inline financial-templates-lib/price-feed/CreatePriceFeed.js:6]: So is this the way to import contract ABI's now?

Comment (mrice32) [reply financial-templates-lib/price-feed/CreatePriceFeed.js:6]: That's the way to get the json from other packages. We _should_ also be able to use  `artifacts.require("@umaprotocol/core/SomeContract")`.

File: liquidator/index.js [hunk 2]
```diff
@@ -1,17 +1,20 @@
...
+  ExpiringMultiPartyClient,
+  Networker,
+  Logger,
+  createReferencePriceFeedForEmp,
+  waitForLogger,
+  delay
+} = require("@umaprotocol/financial-templates-lib");
 
 // Truffle contracts
 const ExpiringMultiParty = artifacts.require("ExpiringMultiParty");
```
Comment (nicholaspai) [inline liquidator/index.js:20]: Is there a reason why these three contract imports are still using `artifacts.require()`, even though it seems we can import them via `require("@umaprotocol/core/build/contracts/x)`?

Comment (mrice32) [reply liquidator/index.js:20]: I talked about this in the description. This refactor will be pretty painful (IMO), and I'd prefer to just let truffle do the work, especially in the tests.

Comment (nicholaspai) [reply liquidator/index.js:20]: Ah I see now

**Comment by mrice32** [comment]
> This is pretty awesome! I really like being able to deal with imports like this.
> Few questions:
> 
> 1. Do we wait until truffle implements their feature in the PR you linked or do we go ahead with the rest of the reactors?
> 2. How does test running work after this PR? I'm guess this has not changed and still is run from the `core` directory?
> 3. At what point do we start moving things into a `packages` directory?

1. We can move forward with the refactors. We just need to wait to run truffle commands outside of core.
2. Yep, that's correct. I expect the release to be in about a week. truffle releases pretty frequently from what I understand.
3. Nick and I decided it makes sense to do any repo directory reorganization after the lerna conversion. That way, we won't have to update the includes twice since they should all look like `require("@umaproject/financial-templates-lib")`, which is non-directory dependent.

**Pull request merged**